modules/simple_forms_api/app/services/simple_forms_api/form_remediation/s3_client.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'fileutils'
+require 'simple_forms_api/form_remediation/configuration/base'
+
+# Built in accordance with the following documentation:
+# https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/platform/practices/zero-silent-failures/remediation.md
+module SimpleFormsApi
+  module FormRemediation
+    class S3Client
+      include FileUtilities
+
+      class << self
+        def fetch_presigned_url(id, type: :submission)
+          new(id:).s3_generate_presigned_url(s3_upload_file_path, type:)
+        end
+      end
+
+      def initialize(config: Configuration::Base.new, type: :remediation, **options)
+        @upload_type = type
+        @config = config
+        @parent_dir = config.parent_dir
+        @presign_s3_url = config.presign_s3_url
+        @temp_directory_path = config.temp_directory_path
+
+        @file_path = options[:file_path]
+        @id = options[:id]
+
+        @archive_path = build_archive!(config:, type:, **options)
+      rescue => e
+        config.handle_error("#{self.class.name} initialization failed", e)
+      end
+
+      def upload
+        config.log_info("Uploading #{upload_type}: #{id} to S3 bucket")
+
+        upload_to_s3
+        cleanup(s3_upload_file_path)
+
+        presign_s3_url ? s3_generate_presigned_url(s3_get_presigned_path) : id
```

Comment:
Just so I understand, if we're just fetching a URL it'll go to `id` here but if we have to generate AND return it, it'll go to `s3_generate_presigned_url`?

Reply:
Presigning URLs can be time consuming at times and if this is being run as a bulk job then it may not even make sense for the user to have presigned urls to click on. I was able to download each one no problem now that I have S3 access and using aws cli, one can download multiple files with one command. All that to say, teams using this service will probably not even want to generate the URLs given how cumbersome and slow it can be. This will instead just print off the ids of the submissions which were remediated instead of each URL.